server-base/src/main/java/org/apache/kylin/rest/service/CubeService.java

```diff
@@ -550,7 +558,30 @@ public CubeInstance deleteSegment(CubeInstance cube, String segmentName) throws
             logger.warn(String.format(Locale.ROOT, msg.getDELETE_SEGMENT_CAUSE_GAPS(), cube.getName(), segmentName));
         }
 
-        return CubeManager.getInstance(getConfig()).updateCubeDropSegments(cube, toDelete);
+        CubeInstance cubeInstance = CubeManager.getInstance(getConfig()).updateCubeDropSegments(cube, toDelete);
+
+        cleanSegmentStorage(Collections.singletonList(toDelete));
+
+        return cubeInstance;
+    }
+
+    // clean segment data in hbase and hdfs
+    private void cleanSegmentStorage(List<CubeSegment> toRemoveSegs) throws IOException {
+        if (KylinConfig.getInstanceFromEnv().cleanStorageAfterDelOperation()) {
```

Comment:
If not enabled, directly return, to make the code more clear (less embed logic);